content/docs/user-guide/troubleshooting.md

```diff
@@ -125,3 +125,39 @@ and retry the DVC command. Specifically, one of:
 
 [internal directories]:
   https://dvc.org/doc/user-guide/project-structure/internal-files
+
+## DVC experiments may fail in shallow Git repositories {#git-shallow}
+
+DVC experiments rely on Git features which may not work properly in shallow Git
+repositories. This error commonly occurs in certain CI environments which use
```

Comment:
Not sure everyone will know what we mean by "CI environments" or even what CI is. Maybe at least we can mention some of those common environments, like GitHub Actions and GitLab CI?